C++/include/algorithm/dynamic_programming/coin_change.hpp

```diff
@@ -19,29 +15,36 @@
 #define COIN_CHANGE_HPP
 
 #include <vector>
-using namespace std;
+
+using std::vector;
 
 int coin_change(const vector<int> coin, int number_of_coins, int amount)
 {
+    if(number_of_coins == 0 && amount == 0) {
+        return 1;
+    } else if(number_of_coins == 0 && amount != 0){

```

Comment:
You can remove `&& amount != 0` since it's always true by this point in the code.